benches/benches/bevy_ecs/fragmentation/mod.rs

```diff
@@ -0,0 +1,99 @@
+use bevy_ecs::prelude::*;
+use bevy_ecs::system::SystemState;
+use criterion::*;
+use glam::*;
+use std::hint::black_box;
+
+criterion_group!(fragmentation_benches, iter_frag_empty);
+
+#[derive(Component, Default)]
+struct Table<const X: usize = 0>(usize);
+#[derive(Component, Default)]
+#[component(storage = "SparseSet")]
+struct Sparse<const X: usize = 0>(usize);
+
+fn flip_coin() -> bool {
+    rand::random::<bool>()
+}
+fn iter_frag_empty(c: &mut Criterion) {
+    let mut group = c.benchmark_group("iter_fragmented(4096)_empty");
+    group.warm_up_time(std::time::Duration::from_millis(500));
+    group.measurement_time(std::time::Duration::from_secs(4));
+
+    group.bench_function("table", |b| {
```

Comment:
```suggestion
    group.bench_function("foreach_table", |b| {
```

To follow the convention in the other benchmarks. It would also be nice to see a non-`foreach` variant of this benchmark.